modules/renter/files.go

```diff
@@ -204,15 +204,23 @@ func (r *Renter) FileList() []modules.FileInfo {
 	lockID := r.mu.RLock()
 	defer r.mu.RUnlock(lockID)
 
+	isOffline := func(id types.FileContractID) bool {
+		offline := r.hostContractor.IsOffline(id)
+		contract, exists := r.hostContractor.ContractByID(id)
```

Comment:
should not be calling exported methods while holding a lock , you've got a readlock on the renter right now and you're making exported calls to the contractor